Lib/ufo2ft/featureCompiler.py

```diff
@@ -354,3 +369,32 @@ def buildTables(self):
             table = mtiLib.build(features.splitlines(), self.ttFont)
             assert table.tableTag == tag
             self.ttFont[tag] = table
+
+
+def warn_about_miscased_insertion_markers(
+    ufo_description: str, feaFile: ast.FeatureFile | ast.Block, patterns: set[str]
+) -> None:
+    """Warn the user about potentially mistyped feature insertion markers."""
+
+    patterns_compiled = tuple(
+        (re.compile(pattern), re.compile(pattern, re.IGNORECASE))
+        for pattern in patterns
+    )
+
+    def _inner(block: Any) -> None:
+        for statement in block.statements:
+            if hasattr(statement, "statements"):
+                _inner(statement)
+            elif isinstance(statement, ast.Comment):
```

Comment:
I don't think it's a good idea to traverse the entire FEA tree for _any_ comments that match the pattern, we should only look within the statements that comprise a FeatureBlock, not anywhere else. I think that's where we look for these.

Reply:
we have an `iterFeatureBlocks` fuction in ufo2ft.featureWriters.ast module that may help with that

Reply:
Shall I change it as well?

Reply:
maybe not. the name `findCommentPattern` suggests it finds a comment anywhere, so let's keep it like that